### DISCUSSION TITLE
sql how to add parameter default as show all instead of just a value 1 time

### DISCUSSION THREAD

**Discussion by Pierrelai99** [Budibase/budibase#4450] [Help]
example: 
select member_reg_date,member_uid
from member 
where member_uid = '{{member_id}}' 

I want to add a parameter = member_id, this can be work when fill in the default value 
![image](https://user-images.githubusercontent.com/99549872/153719749-1760bb08-fc20-4e43-b7f8-1fbb054ab723.png)

when the default value is empty then the query cannot be execute and saved, so how should i do to let it show all values when parameter in is empty and when there is a parameter then can be filter

**Comment by SMetriq** [comment]
Hello, not sure about to do it exactly yet but I would use coalesce function in your sql query like :

member_uid = coalesce( '{{member_id}}' , 'xxxx')

Please tell me if it works fine 

Regards,

**Reply by Pierrelai99** [reply]
thanks for replying, this query is used for when data is null then show what oh, now the problem I meet is how to use budibase to let it be when default is null can show all data when create a new query

**Comment by mjashanks** [comment]
Hey @Pierrelai99 

You could do

```sql
select member_reg_date,member_uid
from member
where member_uid = '{{member_id}}' or '{{member_id}}' = ''
```

**Reply by Pierrelai99** [reply]
thanks for replying, I had try this but fail..., when I click run query, it show query results empty, last time 10++days ago,  query below can function, but now cannot, not sure is that because of budibase update 

select member_reg_date,member_uid
from member
where 1=1 
and (member_uid = '{{member_id}}' or '{{member_id}}' = '')

**Reply by mjashanks** [reply]
might be worth using a dummy default value, rather than blank ....

```sql
select member_reg_date,member_uid
from member
where member_uid = '{{member_id}}' 
or '{{member_id}}' = '_DEFAULT_VALUE_'
```

... and setting the `member_id` default value to `_DEFAULT_VALUE_`
